music21/bar.py

```diff
@@ -293,6 +293,7 @@ def direction(self) -> str:
 
         TODO: show how changing direction changes type.
         '''
+        assert self._direction is not None, '_direction unexpectedly None after initialization'
```

Comment:
We've discussed the potential for a bare `assert` to be bewildering, but the reason Mypy needs it is that we have other code that already depends on this not being None, and if it ever wasn't None, it will crash with even more bewildering messages. So there really is no workaround for the user except opening a bug report on GitHub.

Reply:
Or, actually, the user can run `python -O` to skip the assertions, so it's actually the best of both worlds!